Adding recipe for zetteldesk-kb

Vidianos-Giannitsis opened the pull request:
### Brief summary of what the package does

This package is the default set of keybindings for the Zetteldesk package. Its packaged separately to keep it fully opt-in and also because it would introduce an unnecessary dependency on the hydra package which I used to build this.

### Direct link to the package repository

https://github.com/Vidianos-Giannitsis/zetteldesk.el

### Your association with the package

I am the package's author and maintainer

### Relevant communications with the upstream package maintainer

**None needed**

### Checklist

<!-- Please confirm with `x`: -->

- [x] The package is released under a [GPL-Compatible Free Software License](https://www.gnu.org/licenses/license-list.en.html#GPLCompatibleLicenses)
- [x] I've read [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [x] I've used the latest version of [package-lint](https://github.com/purcell/package-lint) to check for packaging issues, and addressed its feedback
- [ ] My elisp byte-compiles cleanly
- [x] `M-x checkdoc` is happy with my docstrings
- [x] I've built and installed the package using the instructions in [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [ ] I have confirmed some of these without doing them

My elisp does not byte-compile cleanly as it gives me an error that the docstrings of the hydra heads are wider than 80 characters. However, as these are automatically generated I don't know how to fix this.

<!-- After submitting, please fix any problems the CI reports. -->

riscy commented:
Just writing to let you know I see this, but I try to limit myself to one PR per author at a time (you have 3 open at the moment).

riscy commented:
Thanks for this.  Very quick first pass --

More feedback on `zetteldesk-kb.el` from [melpazoid](https://github.com/riscy/melpazoid):
```
- zetteldesk-kb.el#L113: Top-level `define-key` can overwrite bindings.  Try: `(defvar my-map (let ((km (make-sparse-keymap))) (define-key ...) km))`
```

`load`-check on each file:
```
Loading zetteldesk-kb.el
  zetteldesk-kb.el:Error: Emacs 27.1 couldn't load:
  (wrong-type-argument arrayp nil)
```

I think this error is due to this form at the bottom of the file:
```elisp
(define-key zetteldesk-map zetteldesk-kb-hydra-prefix 'zetteldesk-main-hydra/body)
```

Vidianos-Giannitsis commented:
Ok, I fixed this. I wasn't aware of the convention on `define-key` but it wasn't hard to fix. The other error was because the prefix key was nil by default to allow for more customisability. But I guess since this is already optional and changing the value is trivial, it can have a default value. Also, to answer for the other PRs, I didn't know there was a limit so I just made PRs for all the extensions. 

My main concern was getting the main package on MELPA so these can be delayed, I don't really mind. If there's any other issue do let me know.

riscy commented:
Thanks for accommodating!  There's no official limit but we do mention more or less the same thing about dividing review time at the [end of CONTRIBUTING.md](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org#waiting-for-reviews-and-taking-feedback-into-account).  Sometimes it can be hard for me to know which one to review first.

Here's the latest --

`zetteldesk-kb.el` with package-lint 20220412.1648:
```
2 issues found:
106:51: error: This key sequence is reserved (see Key Binding Conventions in the Emacs Lisp manual)
111:0: error: "zetteldesk-map" doesn't start with package's prefix "zetteldesk-kb".
```

Vidianos-Giannitsis commented:
> Thanks for accommodating! There's no official limit but we do mention more or less the same thing about dividing review time at the [end of CONTRIBUTING.md](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org#waiting-for-reviews-and-taking-feedback-into-account). Sometimes it can be hard for me to know which one to review first.
Ok, if its there I must have read it when I initially read that document and then forgot. Don't worry I will keep it in mind if I commit anything new. For these, do them in any order, as they are all extensions of the same package it doesn't matter too much in what order they are done.

Also I was in a hurry all week so I fixed this in a rush and forgot to recheck package-lint which had new issues.

I fixed them. I made the keybinding "C-c z" as that's what I personally recommend. What I forgot was that I initially had set it as nil to avoid this issue. But since setting that to nil raised an error I had to set it to sth. For the other issue, its because the map originates in the definition of the `zetteldesk-mode` in zetteldesk.el. When writing that it made more sense to bind it as `zetteldesk-map`. But because I want the keymap zetteldesk-kb acts on to be the keymap of `zetteldesk-mode` I just changed that to `zetteldesk-kb-map`. It starts with the prefix of zetteldesk so it doesn't raise any new issues.

> Loading zetteldesk-kb.el
>   zetteldesk-kb.el:Error: Emacs 27.1 couldn't load:
>   (wrong-type-argument arrayp nil)
(I have seen this error before and it was caused by not giving `zetteldesk-kb-hydra-prefix` a value)

> My elisp does not byte-compile cleanly as it gives me an error that the docstrings of the hydra heads are wider than 80 characters. However, as these are automatically generated I don't know how to fix this.

As I assume you will run into this next, any ideas on this? I haven't had the time to look into it further, but last time I checked it, I couldn't find a solution as these docstrings are auto generated and I can't control it.

Thanks for the help, Cheers

riscy commented:
> However, as these are automatically generated I don't know how to fix this.

Wow, it's news to me that the Emacs 28 byte-compiler now warns about docstring width.  I wonder how many packages this will have an effect on.

I'd personally look at trying to tame these if possible (shorter names, abbreviate some phrasing). But it looks like you can add e.g.
```elisp
;; Local variables:
;; byte-compile-docstring-max-column: 100
;; end:
```
to the end of your file, which will prevent your users from seeing a raft of warnings during installation.  Keeping it at 100 will let you see warnings if they get any longer.

I had a look at some of the longest docstrings and none of them seem too bad, so I don't think this is a blocker.  Thanks!

Vidianos-Giannitsis commented:
Oh, I see, that's interesting info, will include this then. Thanks!
